docs/content/querying/sql.md

```diff
@@ -1,77 +1,328 @@
+|`CASE expr WHEN value1 THEN result1 \[ WHEN value2 THEN result2 ... \] \[ ELSE resultN \] END`|Simple CASE.|
+|`CASE WHEN boolean_expr1 THEN result1 \[ WHEN boolean_expr2 THEN result2 ... \] \[ ELSE resultN \] END`|Searched CASE.|
+|`NULLIF(value1, value2)`|Returns NULL if value1 and value2 match, else returns value1.|
+|`COALESCE(value1, value2, ...)`|Returns the first value that is neither NULL nor empty string.|
 
-try (Connection connection = DriverManager.getConnection(url, connectionProperties)) {
-  try (ResultSet resultSet = connection.createStatement().executeQuery("SELECT COUNT(*) AS cnt FROM data_source")) {
-    while (resultSet.next()) {
-      // Do something
-    }
-  }
-}
-```
+### Unsupported features
 
-Table metadata is available over JDBC using `connection.getMetaData()` or by querying the "INFORMATION_SCHEMA" tables
-(see below).
+Druid does not support all SQL features, including:
 
-Parameterized queries don't work properly, so avoid those.
+- OVER clauses, and analytic functions such as `LAG` and `LEAD`.
+- JOIN clauses, other than semi-joins as described above.
+- OFFSET clauses.
+- DDL and DML.
 
-### Querying with JSON over HTTP
+Additionally, some Druid features are not supported by the SQL language. Some unsupported Druid features include:
 
-You can make Druid SQL queries using JSON over HTTP by POSTing to the endpoint `/druid/v2/sql/`. The request format
-is:
+- [Multi-value dimensions](multi-value-dimensions.html).
+- [DataSketches aggregators](../development/extensions-core/datasketches-aggregators.html).
+
+## Data types and casts
+
+Druid natively supports four main column types: "long" (64 bit signed int), "float" (32 bit float), "string" (UTF-8
+encoded strings), and "complex" (catch-all for more exotic data types like hyperUnique and approxHistogram columns).
+Timestamps (including the `__time` column) are stored as longs, with the value being the number of milliseconds since 1
+January 1970 UTC.
+
+At runtime, Druid will widen floats to "double" (64 bit float) for certain features, like `SUM` aggregators. But this
+widening is not universal; some floating point operations retain 32 bit precision.
+
+Druid generally treats NULLs and empty strings interchangeably, rather than according to the SQL standard. As such,
+Druid SQL only has partial support for NULLs. For example, the expressions `col IS NULL` and `col = ''` are equivalent,
+and both will evaluate to true if `col` contains an empty string. Similarly, the expression `COALESCE(col1, col2)` will
+return `col2` if `col1` is an empty string. While the `COUNT(*)` aggregator counts all rows, the `COUNT(expr)`
+aggregator will count the number of rows where expr is neither null nor the empty string. String columns in Druid are
+NULLable. Numeric columns are NOT NULL; if you query a numeric column that is not present in all segments of your Druid
```

Comment:
Hmm. It's still not clear to me. Numeric columns are NOT NULL because null numeric values are casted to 0? ```FLOAT NOT NULL``` sounds like that it works like the NOT NULL constraint.